typings/Application.d.ts

```diff
@@ -30,16 +40,12 @@ export interface ApplicationPlugin {
 
 export class ApplicationPlugin implements ApplicationPlugin {}
 
-export class StateManager {
-  addField(name: string, initialValue: any): Property;
-  [key:string]:((name: string, initialValue: string) => Property) | Property
-}
 
-export class Property {
-  constructor(initialValue: any);
-  private _value: any;
+export class Property<T> {
+  constructor(initialValue: T);
+  private _value: T;
   private listeners: Function[];
-  value: any;
+  value: T;
   notifyChange(): void;
   subscribe(callback: Function): void;
```

Comment:
It might be worth doing the typings for the callbacks.
I think it's something like `subscribe(callback: (current?: T, previous?: T) => void): void; `
but I could be wrong.

this way they get the type safety on them.

Reply:
Agree completely.